cuda_core/cuda/core/experimental/_event.py

```diff
@@ -50,19 +51,29 @@ class Event:
 
     """
 
-    __slots__ = ("_handle", "_timing_disabled", "_busy_waited")
+    class _MembersNeededForFinalize:
+        __slots__ = ("handle",)
```

Comment:
Q: We don't need to add `__weakref__` here (meaning a strong reference to `_mnff` will always be held)?

Reply:
Yes, the `Event` object holds a strong reference to `_mnff`, but that is released when the `Event` object goes out of scope (I think from the regular/implicit destructor; I'm not sure what is the correct term).

Based on the behavior of weakref_finalize_toy_example.py shown in the PR description, I believe it works like this:

- The last reference to the `Event` object is released.
- The weakref finalizer runs at that point.
- Then the `Event` object is destroyed, which indirectly also destroys the `_mnff` object.

At no point do we need a weak reference to `_mnff`, therefore we don't have to add `__weakref__` to the `__slots__` of `_MembersNeededForFinalize `.